### ISSUE TITLE
Support code completion for record fields

### ISSUE THREAD

**Issue by AdamSaleh** [dhall-lang/dhall-haskell#1615]
When working with dhall, I often use the `::` operator when defining new records.
It would be really useful if I got suggestions for the fields provided by the type.

**Comment by Gabriel439** [comment]
@AdamSaleh: I agree!  I've also wanted this feature and it seems like an obvious thing to do to take advantage of the user-supplied schema to suggest completions.  I will take a look at this soon if nobody else does
